### PR TITLE
Added angular material design

### PR DESCRIPTION
**Pull request by Kharonus**
Introducing angular material theming with a custom scheme color palettes.

- updated dependencies
- removed boilerplate code from app.component.html
- added material test page with routing
- added multiple example components to test page to verify colors

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -43,4 +45,4 @@
         "tslint": "^5.12.0",
         "typescript": "^3.1.6"
     }
-}
+}
```
Comment (apfohl) [inline package.json]: Please have line breaks at the end of each file. Your editor can help you with that.

File: src/app/app.module.ts [hunk 2]
```diff
@@ -1,20 +1,45 @@
...
 import { AppRoutingModule } from './app-routing.module';
 import { AppComponent } from './app.component';
+import { HeaderComponent } from './components/header/header.component';
+import { MaterialTestComponent } from './components/material-test/material-test.component';
 
 @NgModule({
     declarations: [
-        AppComponent
+        AppComponent,
+        HeaderComponent,
```
Comment (apfohl) [inline src/app/app.module.ts]: It would be nice to have the declarations below the imports.

File: src/xpenses-theme.scss [hunk 3]
```diff
@@ -0,0 +1,126 @@
+// Custom Theming for Angular Material
```
Comment (apfohl) [inline src/xpenses-theme.scss]: How do you came up with these colors?

Comment (Kharonus) [reply src/xpenses-theme.scss]: These are just some randomly created ones from a color scheme generator. Waiting for Hagen to come up with a design. ;)

File: package.json (hunk 1 above)
Comment (Kharonus) [reply package.json]: fixed, sadly I didn't found it on formatting, only on saving in VSC. Does anyone know how to fix it? RIght now, formatting it removes the final line, saving it adds it.

File: src/app/app.module.ts (hunk 2 above)
Comment (Kharonus) [reply src/app/app.module.ts]: fixed

**Comment by apfohl** [comment]
Please consider squash merging.

**Pull request merged**